### PR TITLE
Apply tslint fixes

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
Also add git hook

To use the hook:

```
  git config core.hooksPath git_hooks
```

### CODE REVIEW

File: git_hooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  # Only lint files that were added, copied, modified, or renamed
+  for file in $(git diff --diff-filter=ACMR --cached --name-only . | grep -E ${LINTER_PATTERN}); do
+    local error=0
+
+    git show ":$file" | \
+      ./node_modules/.bin/${LINTER} ${LINTER_ARGS[@]} || error=1
+
+    if [ $error -ne 0 ]; then
+      echo "${LINTER} failed on staged file '$file'"
```
Comment (paddyquinn) [inline git_hooks/pre-commit]: Will this print out the tslint error message? Just saying which files failed is not terribly useful.

File: git_hooks/pre-commit [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  then
+    echo "linter not installed - please run \"npm install\" first"
+  fi
+
+  # Only lint files that were added, copied, modified, or renamed
+  for file in $(git diff --diff-filter=ACMR --cached --name-only . | grep -E ${LINTER_PATTERN}); do
+    local error=0
+
+    git show ":$file" | \
+      ./node_modules/.bin/${LINTER} ${LINTER_ARGS[@]} || error=1
```
Comment (paddyquinn) [inline git_hooks/pre-commit]: I don't quite understand what this line does. Specifically I don't understand what `git show` does when passed a file.

File: git_hooks/pre-commit (hunk 1 above)
Comment (OttoAllmendinger) [reply git_hooks/pre-commit]: the previous command will print the error

File: git_hooks/pre-commit (hunk 2 above)
Comment (OttoAllmendinger) [reply git_hooks/pre-commit]: It shows the staged file. This way we check what is actually about to be commited, which can be different from what is currently on disk.

Comment (paddyquinn) [reply git_hooks/pre-commit]: Cool. Merge away

**Pull request merged**